src/JobCreator.cpp

```diff
@@ -0,0 +1,240 @@
+  for (auto &StreamHDFInfo : StreamHDFInfoList) {
+    try {
+      StreamSettingsList.push_back(
+          extractStreamInformationFromJsonForSource(StreamHDFInfo));
+      Logger->info("Adding stream: {}",
+                   StreamSettingsList.back().ConfigStreamJson);
+      setUpHdfStructure(StreamSettingsList.back(), Task);
+      StreamHDFInfo.InitialisedOk = true;
+    } catch (json::parse_error const &E) {
+      Logger->warn("Invalid json: {}", StreamHDFInfo.ConfigStream);
+      continue;
+    } catch (std::runtime_error const &E) {
+      Logger->warn("Exception while initialising writer module  what: {}  "
+                   "parent: {}  json: {}",
+                   E.what(), StreamHDFInfo.HDFParentName,
+                   StreamHDFInfo.ConfigStream);
+      continue;
+    } catch (...) {
+      Logger->error("Unknown error caught while trying to initialise stream  "
+                    "parent: {}  json: {}",
+                    StreamHDFInfo.HDFParentName, StreamHDFInfo.ConfigStream);
+    }
+  }
+  return StreamSettingsList;
+}
+
+std::unique_ptr<IStreamMaster> JobCreator::createFileWritingJob(
+    StartCommandInfo const &StartInfo,
+    std::shared_ptr<KafkaW::ProducerTopic> const &StatusProducer,
+    MainOpt &Settings) {
+  auto Task =
+      std::make_unique<FileWriterTask>(Settings.ServiceID, StatusProducer);
+  Task->setJobId(StartInfo.JobID);
+  Task->setFilename(Settings.HDFOutputPrefix, StartInfo.Filename);
+
+  logEvent(StatusProducer, StatusCode::Start, Settings.ServiceID, Task->jobID(),
+           "Start job");
+
+  std::vector<StreamHDFInfo> StreamHDFInfoList =
+      initializeHDF(*Task, StartInfo.NexusStructure, StartInfo.UseSwmr);
+
+  std::vector<StreamSettings> StreamSettingsList =
+      extractStreamInformationFromJson(Task, StreamHDFInfoList, Logger);
+
+  if (StartInfo.AbortOnStreamFailure) {
+    for (auto const &Item : StreamHDFInfoList) {
+      // cppcheck-suppress useStlAlgorithm
+      if (!Item.InitialisedOk) {
+        throw std::runtime_error(fmt::format("Could not initialise {}  {}",
+                                             Item.HDFParentName,
+                                             Item.ConfigStream));
+      }
+    }
+  }
+
+  addStreamSourceToWriterModule(StreamSettingsList, Task);
+
+  Settings.StreamerConfiguration.StartTimestamp = StartInfo.StartTime;
+  Settings.StreamerConfiguration.StopTimestamp = StartInfo.StopTime;
+
+  Logger->info("Start time: {}ms",
+               Settings.StreamerConfiguration.StartTimestamp.count());
+  if (Settings.StreamerConfiguration.StopTimestamp.count() > 0) {
+    Logger->info("Stop time: {}ms",
+                 Settings.StreamerConfiguration.StopTimestamp.count());
+  }
+
+  Logger->info("Write file with job_id: {}", Task->jobID());
+  auto s = StreamMaster::createStreamMaster(
+      StartInfo.BrokerInfo.HostPort, std::move(Task), Settings, StatusProducer);
+  s->report(std::chrono::milliseconds{Settings.StatusMasterIntervalMS});
```

Comment:
General comment: We should probably convert `StatusMasterIntervalMS` into a `std::chrono::duration`.

Reply:
Done!